### PR TITLE
add type annotations to kombu/transport/azurestoragequeues

### PR DESCRIPTION
**Pull request by jasonwbarnett**
This isn't complete, but at least it's a start. I'm open to any feedback.

### CODE REVIEW

File: kombu/transport/azurestoragequeues.py
Comment (thedrow) [inline kombu/transport/azurestoragequeues.py:64]: Shouldn't _queue_service be [Optional](https://docs.python.org/3/library/typing.html#typing.Optional)?

File: kombu/transport/azurestoragequeues.py
Comment (thedrow) [inline kombu/transport/azurestoragequeues.py:178]: The type here is Optional too

**Review by thedrow — CHANGES_REQUESTED** [review]
I have one thing I need you to change and then we can merge this PR

File: kombu/transport/azurestoragequeues.py
Comment (jasonwbarnett) [reply kombu/transport/azurestoragequeues.py:64]: I agree, but the pre-commit.ci automatically corrected it and switched it from `Optional[QueueServiceClient]` to this. I just assumed this was an alternative way to declare the same thing, but outside of this project I would always use the `Optional[QueueServiceClient]` notation.

What should I do?

Comment (jasonwbarnett) [reply kombu/transport/azurestoragequeues.py:64]: I made the updates, and I assume pre-commit.ci will add a commit correcting it, but we'll see.

Comment (jasonwbarnett) [reply kombu/transport/azurestoragequeues.py:64]: Yeah, it corrected it so I just re-pushed the original content that you commented on. Should I try to update the pre-commit.ci configuration or merge as is?

**Comment by jasonwbarnett** [comment]
> I have one thing I need you to change and then we can merge this PR

@thedrow I reviewed the inline comments and responded with why I didn't so that. Please let me know how you would like me to proceed.

**Comment by jasonwbarnett** [comment]
I ended up finding the root cause of this. It turns out `pyupgrade` is responsible for making this change, see: https://github.com/asottile/pyupgrade#pep-604-typing-rewrites

**Comment by jasonwbarnett** [comment]
I just rebased this PR off of main.

File: kombu/transport/azurestoragequeues.py
Comment (thedrow) [reply kombu/transport/azurestoragequeues.py:90]: If you can update the pre-commit config that would be great but I wouldn't block merging this PR without that config change.

**Comment by thedrow** [comment]
Yeah let's configure pyupdate to keep `Optional`.

**Comment by jasonwbarnett** [comment]
@thedrow @Nusnus This is ready to be merged.

**Comment by Nusnus** [comment]
LGTM but I'll let @thedrow have the final word here

**Review by auvipy — APPROVED** [review]

**Pull request merged**
